# Limpieza de datos - turno 3: Nadissa

Este notebook ejecuta las reglas aprobadas para `NIVEL`, `SECTOR`, `AREA`, `STATUS`, `MODALIDAD`, `JORNADA` y `PLAN`. Las reglas viven en `src/limpieza.py`; aqui solo se presentan resultados y se genera la salida preliminar reproducible. 

In [1]:
from pathlib import Path
import sys

import pandas as pd
from IPython.display import display

ROOT = Path('..').resolve()
if str(ROOT) not in sys.path:
    sys.path.insert(0, str(ROOT))

from src.diagnostico import cargar_datos
from src.limpieza import CATEGORIAS_NADISSA, limpiar_datos_preliminar
from src.validacion import validar_categorias_nadissa

crudo = cargar_datos()
limpio_pre = limpiar_datos_preliminar(crudo)
validar_categorias_nadissa(limpio_pre)
print(f'Registros conservados: {len(limpio_pre):,}')


Registros conservados: 11,890


## Resultados de las variables asignadas

`AREA = SIN ESPECIFICAR` se convierte a `NA`. Los doce registros de Ciudad Capital con `AREA = RURAL` se conservan. Nadissa verifico manualmente los codigos `00-01-0512-46`, `00-02-0137-46` y `00-07-0007-46`: codigo, nombre y direccion coinciden exactamente con la fuente; los tres pertenecen al area rural.

In [2]:
resumen = pd.DataFrame([
    {'variable': columna, 'tipo_final': str(limpio_pre[columna].dtype),
     'dominio': ', '.join(dominio), 'faltantes': int(limpio_pre[columna].isna().sum())}
    for columna, dominio in CATEGORIAS_NADISSA.items()
])
display(resumen)

capital_rural = limpio_pre[
    limpio_pre['DEPARTAMENTO_ORIGINAL'].eq('CIUDAD CAPITAL')
    & limpio_pre['AREA'].eq('RURAL')
][['CODIGO', 'ZONA_CAPITAL', 'ESTABLECIMIENTO', 'DIRECCION', 'AREA']]
display(capital_rural)


,variable,tipo_final,dominio,faltantes
0,NIVEL,category,DIVERSIFICADO,23
1,SECTOR,category,"PRIVADO, OFICIAL, COOPERATIVA, MUNICIPAL",23
2,AREA,category,"URBANA, RURAL",26
3,STATUS,category,"ABIERTA, CERRADA TEMPORALMENTE, CERRADA DEFINI...",23
4,MODALIDAD,category,"MONOLINGUE, BILINGUE",23
5,JORNADA,category,"DOBLE, VESPERTINA, MATUTINA, SIN JORNADA, NOCT...",23
6,PLAN,category,"DIARIO(REGULAR), FIN DE SEMANA, SEMIPRESENCIAL...",23


,CODIGO,ZONA_CAPITAL,ESTABLECIMIENTO,DIRECCION,AREA
1551,00-01-0512-46,Zona 1,CENTRO EDUCATIVO FUTURISTA -C.E.F.-,13 AVENIDA 4-60,RURAL
2257,00-02-0137-46,Zona 2,"COLEGIO ""ALFA Y OMEGA""",MANZANA 4 LOTE 31 ZONA 0 LOTIFICACION JOYAS DE...,RURAL
2519,00-07-0007-46,Zona 7,INSTITUTO NACIONAL DE EDUCACION DIVERSIFICADA,38 CALLE 33-20 ZONA 7 COLONIA EL AMPARO,RURAL
3209,00-17-0011-46,Zona 17,CENTRO EDUCATIVO DE INFORMATICA SINAI,MANZANA 33 LOTE 6 CANTON CENTRAL ALDEA CANALITOS,RURAL
3210,00-17-0014-46,Zona 17,CENTRO EDUCATIVO DE INFORMATICA SINAI,"MANZANA 33 LOTE 6, CANTON CENTRAL, ALDEA CANAL...",RURAL
3231,00-17-9134-46,Zona 17,CENTRO EDUCATIVO DE INFORMATICA SINAI,CALLE PRINCIPAL MANZANA 33 LOTE 6 CANTON CENTR...,RURAL
3232,00-17-9135-46,Zona 17,CENTRO EDUCATIVO DE INFORMATICA SINAI,MANZANA 33 LOTE 6 CANTON CENTRAL ALDEA CANALITOS,RURAL
3233,00-17-9136-46,Zona 17,CENTRO EDUCATIVO DE INFORMATICA SINAI (POR MAD...,CALLE PRINCIPAL MANZANA 33 LOTE 6 CANTON CENTR...,RURAL
3323,00-18-0262-46,Zona 18,"CENTRO EDUCATIVO BILINGUE ""SAN FRANCISCO DE ASÍS""","SECTOR 1, LOTE 2, ALDEA CONCEPCIÓN SACOJITO",RURAL
3479,00-25-0005-46,Zona 25,COLEGIO MIXTO ABC II,"26 AVENIDA 91-82 ""A"", COLONIA SANTA LUCIA LOS ...",RURAL


## Salida preliminar

La salida conserva las 11,890 filas y todas las columnas de trazabilidad generadas por los turnos anteriores. Se escribe en `data/interim/`, nunca en `data/raw/`.

In [3]:
salida = ROOT / 'data' / 'interim' / 'establecimientos_diversificado_limpio_preliminar.csv'
limpio_pre.to_csv(salida, index=False, encoding='utf-8-sig')
print(f'CSV generado: {salida}')


CSV generado: C:\Users\nadis\Desktop\UVG\semestre 8\Data Science\Proyecto-1-Data-Science\data\interim\establecimientos_diversificado_limpio_preliminar.csv
